enigma-tools-m/src/primitives/km_primitives.rs

```diff
@@ -62,11 +64,8 @@ impl PrincipalMessage {
         }
         let mut to_sign = Vec::with_capacity(3);
         match &self.data {
-            PrincipalMessageType::Request(Some(addresses)) => {
-                to_sign.push(hash::prepare_hash_multiple(addresses.as_ref()));
-            }
             PrincipalMessageType::EncryptedResponse(v) => to_sign.push(v.clone()),
-            PrincipalMessageType::Request(None) => (), // If the request is empty we don't need to sign on it.
+            PrincipalMessageType::Request => (), // If the request is empty we don't need to sign on it.
```

Comment:
I think that this comment is unclear now, we should update it.

Reply:
```suggestion
            PrincipalMessageType::Request => (),
```